src/lib/db/feature-environment-store.ts

```diff
@@ -475,71 +475,81 @@ export class FeatureEnvironmentStore implements IFeatureEnvironmentStore {
     async cloneStrategies(
         sourceEnvironment: string,
         destinationEnvironment: string,
+        projects: string[],
     ): Promise<void> {
         const stopTimer = this.timer('cloneStrategies');
-        const sourceFeatureStrategies = await this.db(
-            'feature_strategies',
-        ).where({
-            environment: sourceEnvironment,
-        });
 
-        const clonedStrategyRows = sourceFeatureStrategies.map(
-            (featureStrategy) => {
-                return {
+        await this.db.transaction(async (trx) => {
+            const sourceFeatureStrategies = await trx(
+                'feature_strategies as fs',
+            )
+                .join('features as f', 'f.name', 'fs.feature_name')
+                .select('fs.*')
+                .where('fs.environment', sourceEnvironment)
+                .whereIn('f.project', projects);
```

Comment:
I think the projects are used as target, not as source:
<img width="574" height="99" alt="image" src="https://github.com/user-attachments/assets/86395dcc-d6bc-4468-a964-b7c0938923f4" />

Reply:
I don't think this is correct, I believe they should be used as both. Copying strategies for projects you haven't selected feels like a bug to me.

There's also no way of doing the same thing for release plans. It is much easier to assume only the selected projects. And release plans and strategies are closely related, so both should follow a similar logic.